### ISSUE TITLE
Consider Token -> any usize-sized type (i.e. wrapped pointer), no Copy requirement

### ISSUE THREAD

**Issue by target-san** [carllerche/mio#340]
Reason: Token requires external associative container to select amongst several of them. Using some kind of wrapped handle would allow a much simpler pointer-like indirection.

Drawbacks: I'm not very familiar with MIO internals, so it might be virtually impossible to allow non-copyable type

**Comment by reem** [comment]
You *can* implement this yourself on top of mio by storing transmuted `Box<T>`s in `Token`, but this is admittedly dangerous to hand roll.

**Comment by target-san** [comment]
The problem is, Token is copy-able. And Slab, where Token resides, is also copy-able. So this heavily depends on whether all this can be changed such that Token is only moved. Unfortunately, I'm not familiar with `mio` intervals to determine this.

**Comment by reem** [comment]
@target-san `Slab` is actually generic over the index type, mio only re-exports a type alias with the index type filled in as `Token`.

**Comment by carllerche** [comment]
I'm going to close this as it Mio requires `Token` to be copyable. This is due to how epoll, etc... work under the hood.

tl;dr, the Mio `EventLoop` cannot take ownership of any data.
